lib/Share/RoomShareProvider.php

```diff
@@ -631,6 +635,11 @@ public function getSharesBy($userId, $shareType, $node, $reshares, $limit, $offs
 	 * @throws ShareNotFound
 	 */
 	public function getShareById($id, $recipientId = null): IShare {
+
+		if (isset($this->sharesByIdCache[$id])) {
```

Comment:
Cache needs to be invalidated on delete/update function calls

Reply:
I have added cache invalidation (cleaning actually :wink:) in all update/delete calls that seemed related in https://github.com/nextcloud/spreed/pull/7743/commits/03f64768cc8219e7718980d7a6eef4d227b83de7.